src/for_2D_build/common/data_type.h

```diff
@@ -36,6 +36,7 @@
 namespace SPH
 {
 using Arrayi = Array2i;
+using DeviceArrayi = DeviceArray2i;
```

Comment:
As you defined DeviceArrayi here, does that means all other variables without "Device" term are HostVaribles?

Reply:
Yes exactly, device variables are using some SYCL built-in datatypes that are optimized for kernel executions. For example DeviceArray2i is `sycl::vec<int,2>`, which will be translated to CUDA vectors at compile-time (for NVIDIA backends for example). I still have not tried it, but it should be possible to set device variables = host variables, it's just that some functions calls would need to be changed (for example Eigen vectors use dot products as `v1.dot(v2)`, while SYCL interface is `sycl::dot(v1, v2)`). 

I already have a branch in which I am trying to set some helper functions that select the right function call depending on the vector's type, something like:
```
auto dot(Eigen::vec v1, Eigen::vec v2) { return v1.dot(v2); }
auto dot(sycl::vec v1, sycl::vec v2) { return sycl::dot(v1, v2); } 
```

So that, within the code, you can just call `dot(v1, v2)` and it would select the right interface.

Reply:
Great! A helper is definitive essential. Looking forward to seeing it in you next implementation.